Those are the ways Rust tests systematically fail. Since they run in parallel and taking into account that memory issues could make tests fail at any time once 

```
running 27 tests
test data_structures::tests::dtls_fingerprint ... ok
test ortc::tests::generate_router_rtp_capabilities_succeeds ... ok
test ortc::tests::generate_router_rtp_capabilities_too_many_codecs ... ok
test ortc::tests::generate_router_rtp_capabilities_unsupported ... ok
test ortc::tests::get_producer_rtp_parameters_mapping_get_consumable_rtp_parameters_get_consumer_rtp_parameters_get_pipe_consumer_rtp_parameters_succeeds ... ok
test ortc::tests::get_producer_rtp_parameters_mapping_unsupported ... ok
test router::audio_level_observer::tests::router_close_event ... ok
test router::active_speaker_observer::tests::router_close_event ... ok
test router::consumer::tests::transport_close_event ... ok
test router::consumer::tests::producer_close_event ... ok
test router::direct_transport::tests::router_close_event ... ok
test router::data_producer::tests::transport_close_event ... ok
test router::data_consumer::tests::transport_close_event ... ok
test router::data_consumer::tests::data_producer_close_event ... ok
test router::pipe_transport::tests::data_producer_close_is_transmitted_to_pipe_data_consumer ... ok
test router::plain_transport::tests::router_close_event ... ok
test router::producer::tests::transport_close_event ... ok
error: test failed, to rerun pass `-p mediasoup --lib`

Caused by:
  process didn't exit successfully: `/home/runner/work/mediasoup/mediasoup/target/debug/deps/mediasoup-8f5b5278ff2f31a1` (signal: 11, SIGSEGV: invalid memory reference)
Error: Process completed with exit code 101.
##[debug]Finishing: cargo test
```

```
running 27 tests
test data_structures::tests::dtls_fingerprint ... ok
test ortc::tests::generate_router_rtp_capabilities_succeeds ... ok
test ortc::tests::generate_router_rtp_capabilities_too_many_codecs ... ok
test ortc::tests::generate_router_rtp_capabilities_unsupported ... ok
test ortc::tests::get_producer_rtp_parameters_mapping_unsupported ... ok
test ortc::tests::get_producer_rtp_parameters_mapping_get_consumable_rtp_parameters_get_consumer_rtp_parameters_get_pipe_consumer_rtp_parameters_succeeds ... ok
test router::audio_level_observer::tests::router_close_event ... ok
test router::active_speaker_observer::tests::router_close_event ... ok
test router::consumer::tests::producer_close_event ... ok
test router::data_consumer::tests::data_producer_close_event ... ok
test router::data_consumer::tests::transport_close_event ... ok
test router::consumer::tests::transport_close_event ... ok
test router::data_producer::tests::transport_close_event ... ok
test router::direct_transport::tests::router_close_event ... ok
test router::plain_transport::tests::router_close_event ... ok
test router::pipe_transport::tests::data_producer_close_is_transmitted_to_pipe_data_consumer ... ok
test router::producer::tests::transport_close_event ... ok
error: test failed, to rerun pass `-p mediasoup --lib`

Caused by:
  process didn't exit successfully: `/home/runner/work/mediasoup/mediasoup/target/debug/deps/mediasoup-8f5b5278ff2f31a1` (signal: 11, SIGSEGV: invalid memory reference)
Error: Process completed with exit code 101.
```

```
running 27 tests
test data_structures::tests::dtls_fingerprint ... ok
test ortc::tests::generate_router_rtp_capabilities_succeeds ... ok
test ortc::tests::generate_router_rtp_capabilities_too_many_codecs ... ok
test ortc::tests::generate_router_rtp_capabilities_unsupported ... ok
test ortc::tests::get_producer_rtp_parameters_mapping_unsupported ... ok
test ortc::tests::get_producer_rtp_parameters_mapping_get_consumable_rtp_parameters_get_consumer_rtp_parameters_get_pipe_consumer_rtp_parameters_succeeds ... ok
test router::active_speaker_observer::tests::router_close_event ... ok
test router::audio_level_observer::tests::router_close_event ... ok
test router::consumer::tests::transport_close_event ... ok
test router::consumer::tests::producer_close_event ... ok
test router::data_consumer::tests::transport_close_event ... ok
test router::direct_transport::tests::router_close_event ... ok
test router::data_producer::tests::transport_close_event ... ok
test router::data_consumer::tests::data_producer_close_event ... ok
test router::plain_transport::tests::router_close_event ... ok
error: test failed, to rerun pass `-p mediasoup --lib`

Caused by:
  process didn't exit successfully: `/home/runner/work/mediasoup/mediasoup/target/debug/deps/mediasoup-8f5b5278ff2f31a1` (signal: 11, SIGSEGV: invalid memory reference)
Error: Process completed with exit code 101.
```

I would like to say that this is pipe transport related but the third test attempt doesn't even execute any "pipe" test.

--- a/node/src/Worker.ts
+++ b/node/src/Worker.ts
@@ -345,7 +345,7 @@ export class Worker<
 		}
 
 		if (disableLiburing) {
-			spawnArgs.push(`--disableLiburing`);
+			spawnArgs.push(`--disableLiburing=true`);
 		}
 
 		logger.debug(

--- a/rust/src/worker.rs
+++ b/rust/src/worker.rs
@@ -414,7 +414,7 @@ impl Inner {
         }
 
         if !enable_liburing {
-            spawn_args.push("--disableLiburing".to_string());
+            spawn_args.push("--disableLiburing=true".to_string());
         }
 
         let id = WorkerId::new();

--- a/worker/src/Settings.cpp
+++ b/worker/src/Settings.cpp
@@ -79,7 +79,7 @@ void Settings::SetConfiguration(int argc, char* argv[])
 	{
 		if (!optarg)
 		{
-			MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
+			MS_THROW_TYPE_ERROR("missing value in command line argument in option '%c'", c);
 		}
 
 		switch (c)

--- a/worker/src/Settings.cpp
+++ b/worker/src/Settings.cpp
@@ -81,6 +81,11 @@ void Settings::SetConfiguration(int argc, char* argv[])
 		{
 			case 'l':
 			{
+				if (!optarg)
+				{
+					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
+				}
+
 				stringValue = std::string(optarg);
 				SetLogLevel(stringValue);
 
@@ -89,6 +94,11 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 't':
 			{
+				if (!optarg)
+				{
+					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
+				}
+
 				stringValue = std::string(optarg);
 				logTags.push_back(stringValue);
 
@@ -97,6 +107,11 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'm':
 			{
+				if (!optarg)
+				{
+					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
+				}
+
 				try
 				{
 					Settings::configuration.rtcMinPort = static_cast<uint16_t>(std::stoi(optarg));
@@ -111,6 +126,11 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'M':
 			{
+				if (!optarg)
+				{
+					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
+				}
+
 				try
 				{
 					Settings::configuration.rtcMaxPort = static_cast<uint16_t>(std::stoi(optarg));
@@ -125,6 +145,11 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'c':
 			{
+				if (!optarg)
+				{
+					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
+				}
+
 				stringValue                                 = std::string(optarg);
 				Settings::configuration.dtlsCertificateFile = stringValue;
 
@@ -133,6 +158,11 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'p':
 			{
+				if (!optarg)
+				{
+					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
+				}
+
 				stringValue                                = std::string(optarg);
 				Settings::configuration.dtlsPrivateKeyFile = stringValue;
 
@@ -141,6 +171,11 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'W':
 			{
+				if (!optarg)
+				{
+					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
+				}
+
 				stringValue = std::string(optarg);
 
 				if (stringValue != Settings::configuration.libwebrtcFieldTrials)

--- a/worker/src/Settings.cpp
+++ b/worker/src/Settings.cpp
@@ -60,7 +60,7 @@ void Settings::SetConfiguration(int argc, char* argv[])
 		{ "dtlsCertificateFile",  optional_argument, nullptr, 'c' },
 		{ "dtlsPrivateKeyFile",   optional_argument, nullptr, 'p' },
 		{ "libwebrtcFieldTrials", optional_argument, nullptr, 'W' },
-		{ "disableLiburing",      no_argument,       nullptr, 'd' },
+		{ "disableLiburing",      optional_argument, nullptr, 'd' },
 		{ nullptr,                0,                 nullptr,  0  }
 	};
 	// clang-format on
@@ -77,15 +77,15 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 	while ((c = getopt_long_only(argc, argv, "", options, &optionIdx)) != -1)
 	{
+		if (!optarg)
+		{
+			MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
+		}
+
 		switch (c)
 		{
 			case 'l':
 			{
-				if (!optarg)
-				{
-					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
-				}
-
 				stringValue = std::string(optarg);
 				SetLogLevel(stringValue);
 
@@ -94,11 +94,6 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 't':
 			{
-				if (!optarg)
-				{
-					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
-				}
-
 				stringValue = std::string(optarg);
 				logTags.push_back(stringValue);
 
@@ -107,11 +102,6 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'm':
 			{
-				if (!optarg)
-				{
-					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
-				}
-
 				try
 				{
 					Settings::configuration.rtcMinPort = static_cast<uint16_t>(std::stoi(optarg));
@@ -126,11 +116,6 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'M':
 			{
-				if (!optarg)
-				{
-					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
-				}
-
 				try
 				{
 					Settings::configuration.rtcMaxPort = static_cast<uint16_t>(std::stoi(optarg));
@@ -145,11 +130,6 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'c':
 			{
-				if (!optarg)
-				{
-					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
-				}
-
 				stringValue                                 = std::string(optarg);
 				Settings::configuration.dtlsCertificateFile = stringValue;
 
@@ -158,11 +138,6 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'p':
 			{
-				if (!optarg)
-				{
-					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
-				}
-
 				stringValue                                = std::string(optarg);
 				Settings::configuration.dtlsPrivateKeyFile = stringValue;
 
@@ -171,11 +146,6 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'W':
 			{
-				if (!optarg)
-				{
-					MS_THROW_TYPE_ERROR("missing value in command line argument: %s", optarg);
-				}
-
 				stringValue = std::string(optarg);
 
 				if (stringValue != Settings::configuration.libwebrtcFieldTrials)
@@ -192,7 +162,12 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 			case 'd':
 			{
-				Settings::configuration.liburingDisabled = true;
+				stringValue = std::string(optarg);
+
+				if (stringValue == "true")
+				{
+					Settings::configuration.liburingDisabled = true;
+				}
 
 				break;
 			}

